pennylane/circuit_graph.py

```diff
@@ -469,9 +470,59 @@ def get_depth(self):
                 self._operation_graph = self._graph.subgraph(
                     list(self._graph.nodes().index(node) for node in self.operations)
                 )
-                self._depth = rx.dag_longest_path_length(self._operation_graph) + 1
+                self._extend_graph(self._operation_graph)
+                self._depth = (
+                    rx.dag_longest_path_length(
+                        self._operation_graph, weight_fn=lambda _, __, w: self._weight_func(w)
+                    )
+                    + 1
+                )
         return self._depth
 
+    @staticmethod
+    def _weight_func(weight):
+        """If weight is a number, use it!"""
+        if isinstance(weight, (int, float)):
+            return weight
+        return 1
+
+    def _extend_graph(self, graph: rx.PyDiGraph) -> rx.PyDiGraph:
+        """Extend graph to account for custom depth operations"""
+        custom_depth_node_dict = {}
+        for op in self.operations:
+            if isinstance(op, ResourcesOperation) and (d := op.resources().depth) > 1:
+                custom_depth_node_dict[graph.nodes().index(op)] = d
+
+        def _link_graph(source_index, target_index, sub_graph, node_index):
+            """Link incoming and outgoing edges for the initial node to the sub-graph"""
+            if target_index == node_index:
+                return sub_graph.nodes().index(f"{node_index}.0")
+            if source_index == node_index:
+                return sub_graph.nodes().index(f"{node_index}.1")
+            return None
+
+        for node_index, depth in custom_depth_node_dict.items():
+            # Construct sub_graph:
+            sub_graph = rx.PyDiGraph()
+            source_node, target_node = (f"{node_index}.0", f"{node_index}.1")
+
+            sub_graph.add_node(source_node)
+            sub_graph.add_node(target_node)
+
+            sub_graph.add_edge(
+                sub_graph.nodes().index(source_node),
+                sub_graph.nodes().index(target_node),
+                depth - 1,  # set edge weight as 1 - depth
+            )
+
+            graph.substitute_node_with_subgraph(
+                node_index,
+                sub_graph,
+                lambda s, t, _: _link_graph(
+                    s, t, sub_graph, node_index
```

Comment:
Why `_` is needed here?

Reply:
The function `substitute_node_with_subgraph` expects its 3rd argument to be a function which takes 3 inputs. Not specifying this will cause an error.